sandbox/modules/identity/src/main/java/org/opensearch/identity/rest/user/resetpassword/ResetPasswordRequest.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.rest.user.resetpassword;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.xcontent.ToXContentObject;
+import org.opensearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+import static org.opensearch.action.ValidateActions.addValidationError;
+
+/**
+ * Request to reset an internal user password
+ */
+public class ResetPasswordRequest extends ActionRequest implements ToXContentObject {
+
+    private String username;
+    private String oldPassword;
```

Comment:
We shouldn't require the old password, if a user forgot there password how would they get a new password issued?

Reply:
Let's leave this as a prototype design for now. (I just wanna transfer our conversation on github :)) I'm not gonna resolve this comment, so that I can use this as a reminder for myself, in case if we wanna change it in the future.